### ISSUE TITLE
Download all button on publicly shared albums

### ISSUE THREAD

**Issue by shalak** [pulsejet/memories#736]
**Is your feature request related to a problem? Please describe.**
Currently, as far as I see, the only way to download whole publicly shared album is to manually click all photos/videos one by one and hit the Download button.

**Describe the solution you'd like**
There should be a button to download whole album. Nice to have would be also a ways to download photos only & videos only.

**Comment by pulsejet** [comment]
Album or folder? Public albums can already be downloaded at once.

**Comment by shalak** [comment]
I'm not sure if I'm aware of a difference between those two.

What I usually do is create a public share on a folder with e.g. photos from holidays (which has sub-folders - each having pictures taken by different people) using the Files app and then prepend the URL path with `apps/memories/`, e.g.:
`https://cloud.example.org/s/SHARE_ID`
->
`https://cloud.example.org/apps/memories/s/SHARE_ID`

Then I send our the memories link and it has one timeline with all the pictures. And there's no "Download all" button on that view.

**Comment by pulsejet** [comment]
What you're describing is a folder share :)
The "albums" section in memories also supports sharing.

**Comment by tryptophane** [comment]
I also miss this feature when sharing a folder (not an album). In a web browser on a pc, it is possible to mark the first picture as selected, and then click on the last picture while holding the Shift key. This will select all files in the folder, which can then be downloaded. This is not possible on a mobile phone, as you can't hold the Shift key.

So I actually miss 2 features:
- a "download all" button on a shared folder
- a "select all" button after having selected at least one file

**Comment by retnag** [comment]
Yes I also miss the download all button. When I share a folder with someone, It is very nice, to have the beautiful preview wall of images that memories provides, but the download all / select all buttons would be a nice addition. I imagine one of the first things the person I share it with wants to do, is download the images.

**Comment by tobiaslangner** [comment]
+1 here. My principal way of sharing photos is through folder sharing.

**Comment by siccovansas** [comment]
This would indeed be really nice to have. I expected a 'Download all files' button to be there as this button is also shown when you share the folder using the normal Nextcloud Files app. So a workaround for now is to also share the Nextcloud Files share URL to your contact if they want to download all files. The `share_id` is the same so to do this you can simply edit the Memories share URL from

    https://<nextcloud_domain>/apps/memories/<share_id>

to

    https://<nextcloud_domain>/s/<share_id>.

**Comment by scottrbaxter** [comment]
Came here looking for exactly this. TBH, i don't really see the value or appeal of separating albums with folders. Considering that creating an Album within Memories requires several additional steps, I can understand why users simply moving a medial folder into their "Folders Path" will simply gravitate to the "Folders" section within Memories, I did as well. I guess realistically, Albums are logical collections of photos from the greater library, not actually dedicated files within a Folder on Nextcloud.

Given this, I'd also like to request a couple of features implemented.
1. Include a "Download" button for and folders under Memories/Folders, as we see with Memories/Albums, today.
2. Automatically create an "Album" for any folder added/created in the Memories "Folders Path". Would be great if this could be set to only be a top level folder (i.e. maybe not ideal to include recursive folders), and also would be nice to have an option to "remove Album" if/when a folder is removed.

Thanks again for this great photo sharing solution and all of your hard work!

**Comment by gouverneurp** [comment]
A possible solution for downloading all files from publicly shared folders can be found here:
https://github.com/pulsejet/memories/pull/1552

**Comment by gouverneurp** [comment]
Can be closed because PR #1552 (change log: https://github.com/pulsejet/memories/commit/6b177961310cbef6f9f14d4f72b3aec1afda722a) has been merged.
